lib/fs-watch/src/lib.rs

```diff
@@ -193,8 +193,12 @@ pub mod inotify {
 
     impl WatchStream {
         pub fn new(paths: Vec<PathBuf>) -> Result<Self, io::Error> {
+            // XXX: this is unfortunately now necessary due to
+            // https://github.com/inotify-rs/inotify/issues/120
+            static mut BUF: [u8; 32] = [0; 32];
+
             let mut inotify = Inotify::init()?;
-            let stream = inotify.event_stream();
+            let stream = inotify.event_stream(unsafe { &mut BUF });
```

Comment:
I understand that this is just cargo-culted from their example, but I don't understand why, in either case, this `unsafe` is okay.

Reply:
I...don't actually think this is safe --- AFAICT, multiple calls to `WatchStream::new` will cause them to share one buffer. This won't hurt us too bad since we don't make multiple `WatchStream`s currently, but I don't like the potential gotcha.

I kind of really hate that API in `inotify` and am a bit mad that they hacked up my code (which was _fine_) to basically require you to do this.

Let me see if there's a better workaround.

Reply:
As of 9d9884ed8ffdb23962f70ed6b5983d52cb62b7bf, every `WatchStream` gets its own buffer on the heap and all is right in the world once again.

Reply:
Oh derp never mind, the only reason that branch built for me locally is because  this code is only built on Linux. hmm, I am not sure if there is a reasonable safe solution for this...